test/e2e/cluster_samples_operator_test.go

```diff
@@ -160,8 +168,22 @@ func verifyConditionsCompleteSamplesAdded(t *testing.T) error {
 			t.Logf("%v", err)
 			return false, nil
 		}
-		if cfg.ConditionTrue(samplesapi.SamplesExist) &&
-			cfg.ConditionTrue(samplesapi.ConfigurationValid) &&
+		if len(cfg.Status.Architectures[0]) == 0 {
+			return false, nil
+		}
+		switch cfg.Status.Architectures[0] {
+		case samplesapi.AMDArchitecture:
+			fallthrough
+		case samplesapi.X86Architecture:
+			if !cfg.ConditionTrue(samplesapi.SamplesExist) {
+				return false, nil
+			}
+		default:
+			if cfg.ConditionTrue(samplesapi.SamplesExist) {
+				return false, fmt.Errorf("samples should not exist on non-x86 arch %s", string(cfg.Status.Architectures[0]))
```

Comment:
this is interesting...  effectively i guess it's true (there will be no samples to exist on non-x86) but how is this condition set?  or in this case, how is it "not" being set?  because there were no samples files to process, we never trip the logic that sets this condition?

And is not setting this condition going to have any downstream effects where we expect to see samples exist set to be true at some point (ie are we going to keep retrying something, or potentially set ourselves to degraded, or some other undesirable side effect of samplesexist remaining "false" forever?)

Reply:
It starts off as false, and then after the api obj creates run without api error, it gets set to true
the api obj creates with our current approach will not be initiated, and subsequently no check to set `SamplesExists` to true will occur.

The fact that we are management state `Removed` should short circuit any initial tries to create, or any retries to create.  And we already force Available==true, progressing==false, degraded==false when we are in `Removed` state.

Reply:
hm, i guess i forgot we are bootstrapping to "removed" on these architectures.

Reply:
I think we've gone back and forth on it @bparees, and in looking at the enhancements doc, we did settle on not doing it.

But when I started messing with the implementation, it became apparent to me that bootstrapping as Removed saved some churn and unnecessary processing within the operator.

That said, you did previously note if the cluster admin does switch to Managed, we have to make sure we do not try to install bad samples.

In general, the lack of files found in the samples operator image's payload assures we don't create any samples.

But I forgot to short circuit the setting of `SamplesExists` to true in the Managed/non-x86 case.
Doing that now.

I should also update the enhancements doc to clear things up, assuming we agree on my slight shift here.

Reply:
i have a general preference for *not* bootstrapping to removed as it makes it easier for us to deliver content to those clusters on upgrades in the future.

if we bootstrap them to removed, we can't change that during an upgrade (we can't tell if they chose it or we did, so we can't change it).

if we bootstrap them to managed w/ no content, we can always add new content to the operator in the future and on upgrade, the operator will install it.

so all in all i'd like to pursue a path where the operator is bootstrapped as "managed" in all cases.

Reply:
ok switched back to "managed" in all cases, with the appropriate short circuit to prevent install attempts but ensure available==true, progressing==false,degraded==false